experimenter/experimenter/nimbus_ui_new/forms.py

```diff
@@ -106,6 +106,109 @@ def clean(self):
         return cleaned_data
 
 
+class NimbusExperimentCloneFormMixin:
+    def __init__(self, *args, parent_slug=None, branch_slug=None, **kwargs):
+        self.parent_slug = parent_slug
+        self.branch_slug = branch_slug
+        super().__init__(*args, **kwargs)
+
+    def clean_name(self):
+        name = self.cleaned_data["name"]
+        slug = slugify(name)
+        if not slug:
+            raise forms.ValidationError(NimbusUIConstants.ERROR_NAME_INVALID)
+        if NimbusExperiment.objects.filter(slug=slug).exists():
+            raise forms.ValidationError(
+                NimbusUIConstants.ERROR_NAME_MAPS_TO_EXISTING_SLUG
+            )
+        return name
+
+    def clean(self):
+        cleaned_data = super().clean()
+        if "name" in cleaned_data:
+            cleaned_data["slug"] = slugify(cleaned_data["name"])
+        return cleaned_data
+
+    def get_changelog_message(self):
+        parent_experiment = NimbusExperiment.objects.get(slug=self.parent_slug)
+        return f"{self.request.user} cloned this experiment from {parent_experiment.name}"
+
+
+class NimbusExperimentSidebarCloneForm(
+    NimbusExperimentCloneFormMixin, NimbusChangeLogFormMixin, forms.ModelForm
+):
+    owner = forms.ModelChoiceField(
+        User.objects.all(),
+        widget=forms.widgets.HiddenInput(),
+    )
+    name = forms.CharField(
+        required=True, widget=forms.TextInput(attrs={"class": "form-control"})
+    )
+    slug = forms.CharField(
+        required=False,
+        widget=forms.widgets.HiddenInput(),
+    )
+
+    class Meta:
+        model = NimbusExperiment
+        fields = ["owner", "name", "slug"]
+
+    def save(self):
+        parent_experiment = NimbusExperiment.objects.get(slug=self.parent_slug)
+        clone = parent_experiment.clone(
+            self.cleaned_data["name"], self.cleaned_data["owner"]
+        )
+        return clone
+
+
+class NimbusExperimentPromoteToRolloutForm(
+    NimbusExperimentCloneFormMixin, NimbusChangeLogFormMixin, forms.ModelForm
+):
+    owner = forms.ModelChoiceField(
+        User.objects.all(),
+        widget=forms.widgets.HiddenInput(),
+    )
+    name = forms.CharField(
+        required=True, widget=forms.TextInput(attrs={"class": "form-control"})
+    )
+    slug = forms.CharField(
+        required=False,
+        widget=forms.widgets.HiddenInput(),
+    )
+
+    class Meta:
+        model = NimbusExperiment
+        fields = ["owner", "name", "slug"]
+
+    def save(self):
+        parent_experiment = NimbusExperiment.objects.get(slug=self.parent_slug)
+        clone = parent_experiment.clone(
+            self.cleaned_data["name"], self.cleaned_data["owner"], self.branch_slug
+        )
+        return clone
+
+
+class ToggleArchiveForm(NimbusChangeLogFormMixin, forms.ModelForm):
+    owner = forms.ModelChoiceField(
+        User.objects.all(),
+        widget=forms.widgets.HiddenInput(),
+    )
```

Comment:
Do we need owner to archive? 🤔

Reply:
Oohhhh, down below you set `owner` on the actual experiment to the archiving user.  Oh I don't think we should do that.  Where did you get that idea?